public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/helpers.tsx

```diff
@@ -44,18 +47,40 @@ const getFeatureNameFromVisParams = (id, seriesParams) => {
 };
 
 function visAggregationToFormik(value) {
-  return [
-    {
-      label: value.params.field.name,
-      type: 'number',
-    },
-  ];
+  if (Object.values(value.params).length !== 0) {
+    return [
+      {
+        label: value.params?.field?.name,
+        type: value.type,
+      },
+    ];
+  }
+  // for count type of vis, there's no field name in the embeddable schema
+  return [];
 }
 
 function visAggregationQueryToFormik(value, seriesParams) {
-  return {
-    [snakeCase(getFeatureNameFromVisParams(value.id, seriesParams))]: {
-      sum: { field: value.params.field.name },
-    },
-  };
+  if (Object.values(value.params).length !== 0) {
+    return {
+      [snakeCase(getFeatureNameFromVisParams(value.id, seriesParams))]: {
+        [visAggregationTypeToFormik(value)]: {
+          field: value.params?.field?.name,
+        },
+      },
+    };
+  }
+  // for count type of vis, there's no field name in the embeddable schema
```

Comment:
same as https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/504/files#r1222290898

Reply:
updated to embeddable-vis schema